src/osd/OSDMap.cc

```diff
@@ -7330,6 +7333,20 @@ void OSDMap::check_health(CephContext *cct,
       ss << down_in_osds.size() << " osds down";
       auto& d = checks->add("OSD_DOWN", HEALTH_WARN, ss.str(),
 			    down_in_osds.size());
+      bool crimson_build = false;
+      bool crimson_pool_is_configured = false;
+      #ifdef WITH_SEASTAR
+      bool crimson_build = true;
+      bool crimson_pool_is_configured = cmd_getval_or<bool>(cmdmap, "crimson", false) ||
+               cct->_conf->get_val<bool>("osd_pool_default_crimson");
+      #endif
+      if (crimson_build && !crimson_pool_is_configured) {
```

Comment:
I tried to do this with one boolean, which is still possible, but I made it as two for the moment for two main reasons:

- clarity of code and to communicate my reasoning
- when using only one boolean, I found it hard to keep a descriptive name and also have it make sense in its default value if crimson is not enabled (I would have had to set the variable as true when WITH_SEASTAR isn't true, which, with the naming, made no sense